fdb-record-layer-lucene/fdb-record-layer-lucene.gradle

```diff
@@ -26,21 +26,25 @@ if (!hasProperty('coreNotStrict')) {
 
 def coreProject = ":${ext.coreProjectName}"
 dependencies {
-    compile project(coreProject)
-    compile "org.apache.lucene:lucene-core:${luceneVersion}"
-    compile "org.apache.lucene:lucene-queryparser:${luceneVersion}"
-    compile "org.apache.lucene:lucene-analyzers-common:${luceneVersion}"
-    compile "org.apache.lucene:lucene-suggest:${luceneVersion}"
-    compile "org.apache.lucene:lucene-highlighter:${luceneVersion}"
-    testCompile project(path: coreProject, configuration: 'tests')
+    implementation project(coreProject)
+    implementation "org.apache.lucene:lucene-core:${luceneVersion}"
+    implementation "org.apache.lucene:lucene-queryparser:${luceneVersion}"
+    implementation "org.apache.lucene:lucene-analyzers-common:${luceneVersion}"
+    implementation "org.apache.lucene:lucene-suggest:${luceneVersion}"
+    implementation "org.apache.lucene:lucene-highlighter:${luceneVersion}"
+    implementation "org.apache.commons:commons-lang3:${commonsLang3Version}"
```

Comment:
I _think_ this is right (that the `lucene` dependencies are supposed to be implementation only and not leak as a part of the API), but there's some risk here that we accidentally exposed something we shouldn't have. I'm fine with leaving this as `implementation`, though, and then fixing that later if we find something is amiss.

Reply:
Nothing seemed to break, so I think we did okay.

Reply:
leaving as-is